src/server/task_manager/task_manager_mixin.test.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+import sinon from 'sinon';
+import { taskManagerMixin } from './task_manager_mixin';
+
+describe('taskManagerMixin', () => {
+  let clock: sinon.SinonFakeTimers;
+  const defaultConfig = {
+    task_manager: {
+      max_workers: 10,
+      override_num_workers: {},
+      index: 'foo',
+      max_attempts: 9,
+      poll_interval: 6000000,
+    },
+  };
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+  });
+
+  afterEach(() => clock.restore());
+
+  test('logs a warning if the elasticsearch plugin is disabled', async () => {
+    const { $test, opts } = testOpts();
+    (opts.server.plugins as any).elasticsearch = undefined;
+    taskManagerMixin(opts.kbnServer, opts.server, opts.config);
+    $test.afterPluginsInit();
+
+    sinon.assert.calledWith(
+      opts.server.log,
+      ['warning', 'task_manager'],
+      'The task manager cannot be initialized when the elasticsearch plugin is disabled.'
+    );
+  });
+
+  test('starts / stops the poller when es goes green / red', async () => {
+    const { $test, opts } = testOpts();
+    taskManagerMixin(opts.kbnServer, opts.server, opts.config);
+    $test.afterPluginsInit();
+
+    const { store, poller } = (opts.server as any).taskManager;
```

Comment:
This looks like it's accessing private fields of the TaskManager instance. How is this working?

Reply:
This is a terrible line of code. That's how. It shouldn't be casting to any... We should update the opts.server type definition... Anyway, TyepScript classes' "private" properties / fields are not really private at run-time, only at compile time.